### DISCUSSION TITLE
how to import image links from an array to grid

### DISCUSSION THREAD

**Discussion by SabSchek** [neuefische/questions#174] [Web] [answered]
Hello again, 

I'm just hitting my comprehension limit again with next.js and really hope someone here can help me.
I would like to create a dynamic grid and display images from my object array. -> see example 
<img width="481" alt="Bildschirmfoto 2022-09-27 um 11 59 50" src="https://user-images.githubusercontent.com/109827469/192496657-6cf4b466-7663-42f2-83f2-0a3e73bbc861.png">

Right now it looks like this, in the next steps I'd like to be able to click on the picture to get to a extra page, but we don't talk about that function now. 
Anyway -> 
<img width="423" alt="Bildschirmfoto 2022-09-27 um 11 53 11" src="https://user-images.githubusercontent.com/109827469/192497005-9b793823-b412-4031-bd81-a41cf9403b1e.png">

I tried several methods and can't find a proper solution to fix it, as it's not working with a local image nor with a external URL.
As always, help is much appreciated.
<img width="1004" alt="Bildschirmfoto 2022-09-27 um 11 53 35" src="https://user-images.githubusercontent.com/109827469/192497255-9a82a4ab-0a44-483c-9547-76ee987bee64.png">
<img width="692" alt="Bildschirmfoto 2022-09-27 um 11 53 45" src="https://user-images.githubusercontent.com/109827469/192497264-9df7addf-b09b-44ed-81e5-a7808932074b.png">
<img width="575" alt="Bildschirmfoto 2022-09-27 um 11 53 53" src="https://user-images.githubusercontent.com/109827469/192497271-f9d8a6fd-76a1-493f-a8f2-0f94ba1262ae.png">

**Comment by MerleMak** [comment]
Hi Fine, maybe try to access the image through src={good.image} instead of goods.image
Let me know if that worked already

**Comment by MerleMak** [comment]
Also the styling of the `<ul>`(the last screenshot)
I think you meant to write 
list-style: none;

Also writing mode: horizontal-lr; I think is producing a bug. You can just delete it :)

**Reply by SabSchek** [reply]
Hi Merle, 
unfortunately, it didn't work, 
<img width="969" alt="Bildschirmfoto 2022-09-27 um 12 14 05" src="https://user-images.githubusercontent.com/109827469/192499734-7a3a754e-59e9-406a-8b64-d178989d8db7.png">

**Comment by MerleMak** [comment, marked as answer]
Hi Fine, 
Okay. I think the reason is, that you need to configure in the `next.config.js` which websites you can add Images from.

So you need to follow [this documentation](https://nextjs.org/docs/basic-features/image-optimization#remote-images)
First you need to add this code  to your  `next.config.js`
```js
module.exports = {
  images: {
    remotePatterns: [
      {
        protocol: 'https',
        hostname: '**.example.com',
      },
    ],
  },
}
```
Obviously the hostname need to be adapted to **.ebay-kleinanzeigen.de :)
Second you need to add the width and height to your image component. 

Hope this does the trick, let me know!

**Comment by MerleMak** [comment]
Can you please send a Pull request, so we can have a closer look 😊
